### ISSUE TITLE
There are 2 empty tab stops after tabbing to the close button

### ISSUE THREAD

**Issue by clane** [vmware/clarity#4086]
## Describe the bug

 [Test URL]
https://clarity-light-theme-v2-a6ykys.stackblitz.io

[WCAG 2.1 Guideline]
2.4.3 Focus Order
(Level A)

[Screen Reader/Browser]
Any

[Description]
There are 2 empty tab stops after tabbing to the close button.

This issue is caused by the presence of tabindex="0" on the following elements.

```
<span class="modal-title-text" tabindex="0">

<div class="modal-body" tabindex="0">
```

[User Impact]
Non-sighted screen reader users and sighted keyboard-only users may be confused by the fact that they can tab to these elements but cannot interact with them. They may think there is a bug because they would only be expecting interactive elements in the tab order and these elements are static.

[Recommendation]
Remove tabindex="0" from any non-interactive elements. In this example the modal body does not contain any content. If the modal body presents a scrollbar on overflow that can be controlled with the keyboard it is considered interactive and this issue does not apply to it.

## How to reproduce

1. Using Safari and the keyboard alone, go to the Test URL
 Tab to the close button
2. Press the Tab key once and observe that the modal title text has focus
3. Press the Tab key once and observe that the modal body has focus
4. Observe that neither of these elements are interactive and should not be in the tab order.

## Expected behavior
Only interactive elements should be in the tab order

## Versions
?**App**

* Angular: Uknown
* Node: : Uknown
* Clarity: : Uknown

**Device:**

* Type: Any
* OS: Any
* Browser: Any
* Version: Any

## Additional notes
NA

**Comment by gnomeontherun** [comment]
@chlane The body has tabindex=0 because it _may_ contain scrollable content (we can't know this, so we always put it there). The title has tabindex=0 because we move focus onto the titles in modals (though in a wizard we do put it in a different place so this one is not necessary in the wizard use case).

The question is what to do, since we can't move focus to the title without it being focusable, so this is a side effect of the implementation we put together for modal focus management, and I don't have a good solution or know if this is a real issue or just an optimization request?

**Comment by chlane** [comment]
I figured the tabindex=0 was there for potentially scrollable content. Since I don't think there is a way t o detect overflow we may have to live with this empty tab stop. But the title should be implemented with tabindex="-1". This allows a static element to be focused by the developer using focus() but not by the user using the Tab key.

**Comment by Shijir** [comment]
@clane we only set focus on the title when the modal dialog opens. Once focus moves to another element, the title will no longer be in focus order. But for the body, the focus is there as it could contain scrollable content. As I don't find the specified issue exist, I am closing this issue.
